Sources/KituraStencil/KituraStencilTemplateEngine.swift

```diff
@@ -41,23 +41,82 @@ public enum StencilTemplateEngineError: Swift.Error {
     case noKeyProvidedForType(value: Encodable)
 }
 
+/**
+ A `TemplateEngine` for Kitura that uses Stencil for templating.
+
+ The default file extension for templates using this engine is `stencil`. If you do
+ not explicitly provide a file extension in the call to `response.render` then this
+ extension will be applied automatically.
+
+ ### Usage Example: ###
+ ```swift
```

Comment:
I think it'd be much clearer if we split the usage example into two as it's confusing here regarding where one example ends and the next starts.

It'd also be nicer if we had an explanation of the second example (with the link to the type-safe templating blog) before the second example code rather than as an afterthought at the end where I think people might miss it.